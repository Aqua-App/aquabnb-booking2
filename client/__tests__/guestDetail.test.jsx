import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import GuestDetail from '../components/guestDetail';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Test suite for GuestDetail component', () => {
  it('it renders without crashing', () => {
    const wrapper = shallow(<GuestDetail />);
  })

  it('should render props guest limit', () => {
    const wrapper = shallow(<GuestDetail guests={7}/>);
    expect(wrapper.find('#limit').text()).toContain('7');
  })

  it('should match snapshot', () => {
    const wrapper = shallow(<GuestDetail />);
    expect(wrapper).toMatchSnapshot();
  })

})