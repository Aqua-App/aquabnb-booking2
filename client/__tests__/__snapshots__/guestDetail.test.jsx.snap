// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for GuestDetail component should match snapshot 1`] = `
<styled.div>
  <styled.div
    style={
      Object {
        "grid-area": "adult",
      }
    }
  >
    Adults
  </styled.div>
  <styled.div
    style={
      Object {
        "grid-area": "adultbutton",
      }
    }
  >
    <styled.button
      onClick={[Function]}
    >
      -
    </styled.button>
    <span
      style={
        Object {
          "align-self": "center",
          "grid-area": "num",
        }
      }
    >
       
    </span>
    <styled.button
      onClick={[Function]}
    >
      +
    </styled.button>
  </styled.div>
  <styled.div
    style={
      Object {
        "grid-area": "child",
      }
    }
  >
    Children
    <styled.div
      style={
        Object {
          "font-size": "12px",
        }
      }
    >
      Ages 2-12
    </styled.div>
  </styled.div>
  <styled.div
    style={
      Object {
        "grid-area": "childbutton",
      }
    }
  >
    <styled.button
      onClick={[Function]}
    >
      -
    </styled.button>
    <span
      style={
        Object {
          "align-self": "center",
          "grid-area": "num",
        }
      }
    >
       
    </span>
    <styled.button
      onClick={[Function]}
    >
      +
    </styled.button>
  </styled.div>
  <styled.div
    style={
      Object {
        "grid-area": "infant",
      }
    }
  >
    Infants
    <styled.div
      style={
        Object {
          "font-size": "12px",
        }
      }
    >
      Under 2
    </styled.div>
  </styled.div>
  <styled.div
    style={
      Object {
        "grid-area": "infantbutton",
      }
    }
  >
    <styled.button
      onClick={[Function]}
    >
      -
    </styled.button>
    <span
      style={
        Object {
          "align-self": "center",
          "grid-area": "num",
        }
      }
    >
       
    </span>
    <styled.button
      onClick={[Function]}
    >
      +
    </styled.button>
  </styled.div>
  <styled.div
    id="limit"
    style={
      Object {
        "grid-area": "note",
      }
    }
  >
     guests maximum. Infants don't count toward the number of guests.
  </styled.div>
  <styled.button>
    Close
  </styled.button>
</styled.div>
`;
